### PR TITLE
Teleport V5

### PR DESCRIPTION
**Pull request by alex-kovoy**
This PR contains Teleport V5.

1. New Navigation.
2. Local User Management.
3. Add Node Dialog that generates auto-install bash script.
4. Add App Dialog that generates auto-install bash scripts.
5. AAP Application Launcher.
6. Brings back the Recorded Sessions feature.

### CODE REVIEW

File: packages/shared/hooks/useAttemptNext.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2019 Gravitational, Inc.
```
Comment (webvictim) [inline packages/shared/hooks/useAttemptNext.ts:2]:
```suggestion
Copyright 2019 Gravitational, Inc.
```
```suggestion
Copyright 2020 Gravitational, Inc.
```

File: packages/teleport/src/AppLauncher/useAppLauncher.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2019 Gravitational, Inc.
```
Comment (webvictim) [inline packages/teleport/src/AppLauncher/useAppLauncher.ts:2]:
```suggestion
Copyright 2020 Gravitational, Inc.
```

File: packages/teleport/src/Apps/AddApp/AddApp.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { Flex } from 'design';
+//import * as Icons from 'design/Icon';
```
Comment (webvictim) [inline packages/teleport/src/Apps/AddApp/AddApp.tsx:19]: Remove comments?

File: packages/teleport/src/Apps/AddApp/AddApp.tsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+import Dialog, { DialogTitle } from 'design/Dialog';
+import useTeleport from 'teleport/useTeleport';
+//import { TabIcon } from 'teleport/components/Tabs';
```
Comment (webvictim) [inline packages/teleport/src/Apps/AddApp/AddApp.tsx:22]: Remove comments?

File: packages/teleport/src/Apps/AddApp/AddApp.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+          title="Automatically"
+          active={automatic}
+          onClick={() => setAutomatic(true)}
+        />
+        <TabIcon
+          Icon={Icons.Cog}
+          title="Manually"
+          active={!automatic}
+          onClick={() => setAutomatic(false)}
+        /> */}
```
Comment (webvictim) [inline packages/teleport/src/Apps/AddApp/AddApp.tsx:68]: Comments

File: packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+                placeholder="https://localhost:4000"
+                onKeyPress={e => handleEnterPress(e, validator)}
+                onChange={e => setUri(e.target.value)}
+              />
+            </Flex>
+            {!cmd && (
+              <Text mb="3">
+                Provide a name and URL of your application to generate a script
+                that will automatically install and configure the App service on
+                the server that can access your application.
```
Comment (webvictim) [inline packages/teleport/src/Apps/AddApp/Automatically/Automatically.tsx:78]:
```suggestion
                Provide the name and URL of your application to generate a script
                that will automatically install and configure the App service on
                the server that can access your application.
```

File: packages/teleport/src/Apps/AddApp/Manually/Manually.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+export default function Manually({ version, onClose }: Props) {
+  return (
+    <>
+      <DialogContent minHeight="240px" flex="0 0 auto">
+        <Box>
+          <Box mb={4}>
+            <Text bold as="span">
+              Step 1
+            </Text>{' '}
+            - Download teleport package on your computer
```
Comment (webvictim) [inline packages/teleport/src/Apps/AddApp/Manually/Manually.tsx:33]:
```suggestion
            - Download Teleport package to your computer
```

File: packages/teleport/src/Apps/Empty/Empty.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <Image width="180px" src={emptyPng} />
+      </Box>
+      <Box>
+        <Box pr={4} mb={6}>
+          <Text typography="h6" mb={3}>
+            SECURE YOUR FIRST APPLICATION
+          </Text>
+          <Text mb={3}>
+            Teleport Application Access provides secure access to internal
+            applications without the need for a VPN but with the audibility and
```
Comment (webvictim) [inline packages/teleport/src/Apps/Empty/Empty.tsx:37]:
```suggestion
            applications without the need for a VPN and with the auditability and
```

File: packages/teleport/src/Clusters/Clusters.tsx [hunk 9]
```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 Gravitational, Inc.
```
Comment (webvictim) [inline packages/teleport/src/Clusters/Clusters.tsx:2]:
```suggestion
Copyright 2020 Gravitational, Inc.
```

File: packages/teleport/src/Clusters/useClusters.ts [hunk 10]
```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 Gravitational, Inc.
```
Comment (webvictim) [inline packages/teleport/src/Clusters/useClusters.ts:2]:
```suggestion
Copyright 2020 Gravitational, Inc.
```

File: packages/teleport/src/AuthConnectors/templates/github.yaml [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+  client_id: <client-id>
+  # Github OAuth app client secret
+  client_secret: <client-secret>
+  # Github will make a callback to this URL after successful authentication
+  # cluster-url is the address the cluster UI is reachable at
+  redirect_url: "https://<cluster-url>/v1/webapi/github/callback"
+  # connector display name that will be appended to the title of "Login with"
+  # button on the cluster login screen so it will say "Login with Github"
+  display: Github
+  # mapping of Github team memberships to Gravity cluster roles
```
Comment (kimlisa) [inline packages/teleport/src/AuthConnectors/templates/github.yaml:16]: nit: capitalize `mapping`

File: packages/teleport/src/AuthConnectors/templates/github.yaml [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  name: new_github_connector
+spec:
+  # Github OAuth app client ID
+  client_id: <client-id>
+  # Github OAuth app client secret
+  client_secret: <client-secret>
+  # Github will make a callback to this URL after successful authentication
+  # cluster-url is the address the cluster UI is reachable at
+  redirect_url: "https://<cluster-url>/v1/webapi/github/callback"
+  # connector display name that will be appended to the title of "Login with"
```
Comment (kimlisa) [inline packages/teleport/src/AuthConnectors/templates/github.yaml:13]: nit: capitalize `connector

File: packages/teleport/src/AuthConnectors/templates/oidc.yaml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+# We recommend using oidc for G Suite, Auth 0 and Keyclock
+#
+kind: oidc
+version: v2
+metadata:
+  name: new_oidc_connector
+spec:
+  redirect_url: "https://<cluster-url>/v1/webapi/oidc/callback"
+  client_id: <client id>
+  # connector display name that will be appended to the title of "Login with"
```
Comment (kimlisa) [inline packages/teleport/src/AuthConnectors/templates/oidc.yaml:12]: nit: capitalize `connector`

File: packages/teleport/src/AuthConnectors/templates/saml.yaml [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+# Example resource for a SAML connector
+# This connector can be used for SAML endpoints like Okta
+#
+kind: saml
+version: v2
+metadata:
+  # the name of the connector
+  name: new_saml_connector
+spec:
+  # connector display name that will be appended to the title of "Login with"
```
Comment (kimlisa) [inline packages/teleport/src/AuthConnectors/templates/saml.yaml:11]: nit: capitalize `connector`

File: packages/teleport/src/AuthConnectors/templates/saml.yaml [hunk 15]
```diff
@@ -0,0 +1,55 @@
+#
+# Example resource for a SAML connector
+# This connector can be used for SAML endpoints like Okta
+#
+kind: saml
+version: v2
+metadata:
+  # the name of the connector
```
Comment (kimlisa) [inline packages/teleport/src/AuthConnectors/templates/saml.yaml:8]: nit: capitalize `the`

**Review by kimlisa — APPROVED** [review]

File: packages/teleport/src/Apps/AddApp/AddApp.tsx (hunk 5 above)
Comment (alex-kovoy) [reply packages/teleport/src/Apps/AddApp/AddApp.tsx:68]: this is temporary, will be uncommented once we enable the Manual section on the Add Application Dialog.

**Pull request merged**